master/buildbot/process/build.py

```diff
@@ -579,6 +580,14 @@ def getSummaryStatistic(self, name, summary_fn, initial_value=_sentinel):
         else:
             return reduce(summary_fn, step_stats_list, initial_value)
 
+    @defer.inlineCallbacks
+    def postProperties(self):
+        # we get all properties, post to MetricsService to filter out
+        # the one that a user wants to post.
+        properties = interfaces.IProperties(self)
+        yield self.master.stats_service.postProperties(properties,
+                                                       self.builder.name)
+
```

Comment:
I think this could be better handled with mq api.
Just register to ("builds", None, "properties", None)
You will need to actually make that event work. because it is not yet handled.
see data/properties.py